app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
```

Comment:
It is the correct logic, see https://github.com/18F/identity-private/issues/1333#issuecomment-277835063. In the current app, if you sign up via LOA3 you won't see a recovery code screen either. I thought that was the intended behavior, to prevent the user from seeing the recovery code twice?